### PR TITLE
[Python] Add (partial) support for `numpy.datetime64` objects

### LINKED ISSUE #5655: Unable to transform python value of type '<class 'numpy.datetime64'>' to DuckDB LogicalType
### What happens?

I'm trying to add support for ibis+duckdb to hvplot/ HoloViews. See https://github.com/holoviz/holoviews/issues/5557#issuecomment-1345157033. When testing crossfiltering between plots I get `duckdb.NotImplementedException: Not implemented Error: Unable to transform python value of type '<class 'numpy.datetime64'>' to DuckDB LogicalType` error.

### To Reproduce

```python
from pathlib import Path

import duckdb
import ibis
import numpy as np
import pandas as pd
import requests

TRIP_DATA_PATH = "yellow_tripdata_2022-02.parquet"

if not Path(TRIP_DATA_PATH).exists():
    url = "https://d37ci6vzurychx.cloudfront.net/trip-data/yellow_tripdata_2022-02.parquet"
    response = requests.get(url, verify=False)
    with open(TRIP_DATA_PATH, "wb") as f:
        f.write(response.content)

DUCKDB_PATH = "DuckDB.db"

if not Path(DUCKDB_PATH).exists():
    duckdb_con = duckdb.connect(DUCKDB_PATH)
    duckdb_con.execute(f"CREATE TABLE big_df AS SELECT * FROM '{TRIP_DATA_PATH}'")
else:
    duckdb_con = duckdb.connect(DUCKDB_PATH)

duckdb_con.execute("select * from big_df where big_df.tpep_pickup_datetime >= CAST(? AS TIMESTAMP WITHOUT TIME ZONE)", parameters=[np.datetime64('2022-02-08T06:01:38.761310')]).execute()
```

### OS:

windows

### DuckDB Version:

0.6.0 (and  0.6.2.dev13)

### DuckDB Client:

python

### Full Name:

Marc Skov Madsen

### Affiliation:

datamodelsanalytics.com

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### Additional Context

As I am originally using duckdb via Ibis, I've also added the issue to the Ibis issues https://github.com/ibis-project/ibis/issues/4997.

I don't know if this is as expected (it is not for me), if Ibis should handle the numpy conversion or if DuckDb should.

### PR DESCRIPTION
**Pull request by Tishj**
This PR fixes #5655 

Numpy uses their own value for datetime64, which just wraps `datetime.datetime` most of the time.
This PR adds partial support for the type, by taking the underlying object.

This is partial because the underlying object could still not be a `datetime.datetime`, for example:
```py
import duckdb
import numpy as np
import datetime

class TestNumpyDatetime64(object):
    def test_numpy_datetime64(self, duckdb_cursor):
        duckdb_con = duckdb.connect()

        duckdb_con.execute("create table tbl(col TIMESTAMP)")
        duckdb_con.execute("insert into tbl VALUES (CAST(? AS TIMESTAMP WITHOUT TIME ZONE))", parameters=[np.datetime64('2022-02-08T06:01:38.76131000')])
        assert [(datetime.datetime(2022, 2, 8, 6, 1, 38, 761310),)] == duckdb_con.execute("select * from tbl").fetchall()
```
causes:
```
Error: Conversion Error: Unimplemented type for cast (BIGINT -> TIMESTAMP)
```

because `np.datetime64('2022-02-08T06:01:38.76131000')` is too big to fit in a `datetime.datetime`, it uses int instead

### CODE REVIEW

File: tools/pythonpkg/tests/fast/pandas/test_df_object_resolution.py [hunk 1]
```diff
@@ -376,6 +376,21 @@ def test_integer_conversion_fail(self, duckdb_cursor):
         double_dtype = np.dtype('object')
         assert isinstance(converted_col['0'].dtype, double_dtype.__class__) == True
 
+    # numpy.datetime64 is just a wrapper around a datetime.datetime object, because reasons..
+    def test_numpy_datetime(self):
+        numpy = pytest.importorskip("numpy")
+
+        standard_vector_size = 2048
+        data = []
+        data += [numpy.datetime64('2022-12-10T21:38:24.578696') for x in range(standard_vector_size)]
```
Comment (Mause) [inline tools/pythonpkg/tests/fast/pandas/test_df_object_resolution.py:385]:
```suggestion
        data += [numpy.datetime64('2022-12-10T21:38:24.578696')] * standard_vector_size
```
Small nitpick, can be applied to the other sites as well :)

File: tools/pythonpkg/tests/fast/pandas/test_df_object_resolution.py [hunk 2]
```diff
@@ -376,6 +376,21 @@ def test_integer_conversion_fail(self, duckdb_cursor):
         double_dtype = np.dtype('object')
         assert isinstance(converted_col['0'].dtype, double_dtype.__class__) == True
 
+    # numpy.datetime64 is just a wrapper around a datetime.datetime object, because reasons..
+    def test_numpy_datetime(self):
+        numpy = pytest.importorskip("numpy")
+
+        standard_vector_size = 2048
```
Comment (Mause) [inline tools/pythonpkg/tests/fast/pandas/test_df_object_resolution.py:383]: Maybe this should be a shared variable somewhere?

Comment (Tishj) [reply tools/pythonpkg/tests/fast/pandas/test_df_object_resolution.py]: I've turned it into a static method on DuckDB and added it to the duckdb python module as `__standard_vector_size__`

**Review by Mytherin — APPROVED** [review]
Thanks for the PR! LGTM. Could you just merge with master?

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**